pkgs/by-name/ba/basalt-monado/package.nix

```diff
@@ -0,0 +1,111 @@
+{
+  stdenv,
+  lib,
+  fetchFromGitLab,
+  fetchFromGitHub,
+  autoPatchelfHook,
+  boost,
+  bzip2,
+  cereal,
+  cli11,
+  cmake,
+  eigen,
+  extra-cmake-modules,
+  fmt,
+  freeglut,
+  glew,
+  libGL,
+  magic-enum,
+  opencv,
+  opengv,
+  pangolin,
+  pkg-config,
+  sophus,
+  tbb,
+  nix-update-script,
+}:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "basalt-monado";
+  version = "release-673cc5c6";
+
+  src = fetchFromGitLab {
+    domain = "gitlab.freedesktop.org";
+    owner = "mateosss";
+    repo = "basalt";
+    rev = finalAttrs.version;
+    hash = "sha256-cj4R6QtZ2yK8VEeNxu5KBNAQez6O0hsTc8CLdIxegyE=";
+  };
+
+  headers-src = fetchFromGitLab {
+    domain = "gitlab.freedesktop.org";
+    owner = "mateosss";
+    repo = "basalt-headers";
+    rev = "28b09b3a5802d44835655778cdd9b1974569bd47";
+    hash = "sha256-gCHCB/1hEy3R/MGexdbq3tIz30Ciw2QmmAqz5wjKhM0=";
+  };
+
+  postUnpack = ''
+    rm -r source/thirdparty/basalt-headers
+    ln -s ${finalAttrs.headers-src} source/thirdparty/basalt-headers
+  '';
+
+  patches = [
+    ./fixed-cmakes.patch
+    ./remove-ros.patch
```

Comment:
That patch doesn't apply

Reply:
I was looking more into the cmake file and found out that there weren't really needed, so now the patches are removed, but now it compiles some of its dependencies from a git submodule. I tested the main binary and everything seems fine, but if we _must_ build the third party dependencies as derivations, I can look into making these patches look better.